src/test/kotlin/wordle/domain/WordComparatorTest.kt

```diff
@@ -0,0 +1,65 @@
+package wordle.domain
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+private val 오늘의_단어 = Word("hello")
+
+class WordComparatorTest {
+    @Test
+    fun `단어의 모든 글자가 '불일치'하다`() {
+        val answerWord = Word("adapt")
+
+        val actual = WordComparator(오늘의_단어).matchCorrect(answerWord).matchPresent(answerWord)
```

Comment:
위에 말씀드린거랑 비슷한 이야기인데 이렇게 "완전일치 먼저검사하고 부분일치 검사를 해야한다"라는 도메인지식이 중복되는 단점도 있는것같아요!

Reply:
게임 알고리즘 상 어쩔수 없는 부분이라 생각하는데 점프님은 어떻게 해결하셨나요??
given 에서 matchCorrect() 를 호출하는 방법이라면 아래처럼 하는 방법과 동일할 것 같습니다!
```suggestion
        // given
        val comparator = WordComparator(오늘의_단어).matchCorrect(answerWord)
        
        // when
        comparator.matchPresent(answerWord)
        
        // then
        assertThat(//....)
```

Reply:
조금 기우일수도 있긴한데요ㅎㅎ 
지금 WordComparator 에서 완전일치와 부분일치의 순서를 강제하지 않고 있기 때문에 다른 개발자가 WordGameLogic 을 거치지않고 바로 `WordComparator().matchPresent().matchCorrect()` 처럼 순서를 바꿔서 오용할 가능성이 있다고 생각했어요!
그리고 만약 여러 클래스에서 단어 비교를 위해서 WordComparator에 바로 접근한다면  `WordComparator().matchCorrect(answerWord).matchPresent(answerWord)` 해당 문장이 반복될 거고 그게 도메인로직(완전->부분 순으로 해야한다) 의 중복이라 생각했어요! 

matchCorrect, matchPresent 를 외부에 노출하지 않고 WordComparator에 순서를 보장한 compare 메서드를 하나 만들것 같아요